closed/src/java.base/share/classes/com/sun/crypto/provider/NativeGaloisCounterMode.java

```diff
@@ -469,39 +493,54 @@ int decryptFinal(byte[] in, int inOfs, int len,
                      byte[] out, int outOfs)
         throws IllegalBlockSizeException, AEADBadTagException,
         ShortBufferException {
+        int bufSize = 0;
+        byte[] keyCopy;
+        byte[] ivCopy;
+        byte[] aad;
+        int tagLenBytes = this.tagLenBytes;
+
+        // need to synchronize this block as if there are concurrent calls to decryptFinal
+        // the shared variables aadBuffer and ibuffer may corrupt input and cause a segfault
+        // in openssl code
         if (len < 0) {
             throw new ProviderException("Input length is negative");
         }
-        if (len < tagLenBytes - ibuffer.size()) {
-            throw new AEADBadTagException("Input too short - need tag");
-        }
-        if (len > MAX_BUF_SIZE - ibuffer.size()) {
-            throw new ProviderException("SunJCE provider only supports "
-                + "a positive input size up to " + MAX_BUF_SIZE + " bytes");
-        }
 
-        if (out.length - outOfs < len  + ibuffer.size() - tagLenBytes) {
-            throw new ShortBufferException("Output buffer too small");
-        }
+        synchronized (this) {
+            bufSize = ibuffer.size();
 
-        byte[] aad = ((aadBuffer == null || aadBuffer.size() == 0) ?
-                       emptyAAD : aadBuffer.toByteArray());
+            if (len < (tagLenBytes - bufSize)) {
+                throw new AEADBadTagException("Input too short - need tag");
+            }
 
-        aadBuffer = null;
+            if (len > (MAX_BUF_SIZE - bufSize)) {
+                throw new ProviderException("SunJCE provider only supports "
+                    + "a positive input size up to " + MAX_BUF_SIZE + " bytes");
+            }
 
-        if (len > 0) {
-            ibuffer.write(in, inOfs, len);
-        }
+            if ((out.length - outOfs) < (len  + bufSize - tagLenBytes)) {
```

Comment:
Please remove extra space between `len` and `+`.

Reply:
The `len +` was there before this PR ([here](https://github.com/ibmruntimes/openj9-openjdk-jdk16/pull/66/files#diff-f10a1f12a17b8a26e7d67158f30e09de1a3e0ad7d71854dfbfb5b358c66931b3L483)), changing this might break unit tests.

Reply:
I'm just asking you to change `len  +` to `len +` (so there's only one space instead of two between the operand and the operator.

Reply:
Ah, I didn't see that. Fixed it now.